src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -40,416 +44,554 @@
-				pw.write("Host: "+thirdpartyHost+"\n\n");
-				pw.flush();
-				
-				//read the response via a 4k buffer
-				ByteArrayOutputStream bos = new ByteArrayOutputStream ();
-				byte [] buffer = new byte [1024*4];
-				int bytesRead = is.read(buffer);
-				int totalBytesRead = 0;
-				while (bytesRead > -1) {
-					totalBytesRead+= bytesRead;					
-					bos.write(buffer, 0, bytesRead);					
-					bytesRead = is.read(buffer);
-				}
-				String response = new String (bos.toByteArray());				
-				if (log.isDebugEnabled()) log.debug("Response is "+totalBytesRead+" bytes: \n"+response);
-				
-				Matcher m = thirdPartyContentPattern.matcher(response);
-				if (m.matches()) {
-					if (log.isDebugEnabled()) log.debug("Response matches expected third party pattern!");
-					is.close();
-					os.close();
-					bos.close();
-					socket.close();
-					return true;
-				} else { 
-					if (log.isDebugEnabled()) log.debug("Response does *not* match expected third party pattern");
-				}
-				is.close();
-				os.close();
-				bos.close();
-				socket.close();
-				return false;				
-			}
-			else  {
-				if (log.isDebugEnabled()) { 
-					log.debug ("Could not establish a socket connection to a third party using the CONNECT HTTP method: NULL socket returned, or non-200 response");
-					log.debug ("The status line returned: "+statusLine);
-				}
-				return false;
-			} 
-		}
-		catch (Exception e) {
-			if (log.isDebugEnabled()) log.debug ("Could not establish a socket connection to a third party using the CONNECT HTTP method", e);
-		}
-		return false;
-	}
+    /* WebDAV methods */
+    private static final String COPY = "COPY";
+    private static final String LOCK = "LOCK";
+    private static final String MKCOL = "MKCOL";
+    private static final String MOVE = "MOVE";
+    private static final String PROPFIND = "PROPFIND";
+    private static final String PROPPATCH = "PROPPATCH";
+    private static final String UNLOCK = "UNLOCK";
+
+    /* These are the 'default' HTTP methods which are considered as insecure */
+    public static final String[] INSECURE_DEFAULT_METHODS = {
+            /* Used by proxy server */
+            HttpRequestHeader.CONNECT,
+            /* This is almost never implemented by the server */
+            HttpRequestHeader.DELETE,
+            /* Request for information about the communication options available */
+            HttpRequestHeader.OPTIONS,
+            /* Request for a change in a resource identified by the URI */
+            HttpRequestHeader.PATCH,
+            /* For putting or updating a resource on the server */
+            HttpRequestHeader.PUT,
+            /* Mostly used for debugging purposes */
+            HttpRequestHeader.TRACE,
+            /* M$ version of TRACE */
+            HttpRequestHeader.TRACK
+    };
+
+    /* These are the WEBDAV methods bundled */
+    public static final String[] WEBDAV_METHODS = {COPY, LOCK, MKCOL, MOVE, PROPFIND, PROPPATCH, UNLOCK};
```

Comment:
- `private`.
 - This should be a `List` (avoids converting it each time when used later).